content/SCALE/SCALETutorials/CommunityRecommends/SpotlightSupportSamba/index.md

```diff
@@ -0,0 +1,171 @@
+---
+title: "Spotlight Support on a SCALE SMB Share"
+description: "This article describes how to configure a SCALE SMB (Samba) share to support the Spotlight search API"
+tags:
+ - scalecommunity
+ - scalesmb
+---
+
+{{< toc >}}
+
+This is a fast spun up tutorial to demonstrate how to have a Samba share on TrueNAS SCALE (in short: TNS) supporting macOS' Spotlight search API. My goal was to have my scans saved inside a network folder being indexed and spotlight enabled. So I write this tutorial for my "scans" share.
+For having this to work we will install an ElasticSearch engine, a script called fscrawler and tesseract libraries and will also show you how you could configure each part of the toolchain to make this work. We will heavily rely on docker images, as I don't want to spin up an extra VM within my VM ;)
+
+## ElasticSearch
+
+or in short within this tutorial only "ES" ([Elastic LINK](https://www.elastic.co)) is an engine that enables you to process searches in an "elastic" way. That means after querying it the search hits will be returned immediately and not after the search was completed. So results will shown may increase after some time, depending on the database ES utilizes. We will use ES 8.4.3 with our docker image
+
+## FS Crawler
+
+is the script that builds the index in the ES database. It can be optimized to index specific values of your files and folders, according to your needs. For example if you prefer to search for titles it mau be better for you to not have a fulltext search enabled. Someone else likes to keep an eye only on the size of the files and wants to search for file and folder size only. If you need more details, feel free to dive deeper into this topic with the fscrawler documentation ([FS Crawler LINK](https://fscrawler.readthedocs.io/en/latest/)). We will use FS Crawler 2.10-SNAPSHOT.
```

Comment:
```suggestion
is the script that builds the index in the ES database. It can be optimized to index specific values of your files and folders, according to your needs. For example if you prefer to search for titles it may be better for you to not have a fulltext search enabled. Someone else likes to keep an eye only on the size of the files and wants to search for file and folder size only. If you need more details, feel free to dive deeper into this topic with the fscrawler documentation ([FS Crawler LINK](https://fscrawler.readthedocs.io/en/latest/)). We will use FS Crawler 2.10-SNAPSHOT.
```